web/src/engine/websites/CuuTruyen.ts

```diff
@@ -0,0 +1,113 @@
+﻿import { Tags } from '../Tags';
+import icon from './CuuTruyen.webp';
+import { Chapter, DecoratableMangaScraper, Manga, Page, type MangaPlugin } from '../providers/MangaPlugin';
+import * as Common from './decorators/Common';
+import { FetchJSON } from '../platform/FetchProvider';
+import type { Priority } from '../taskpool/DeferredTask';
+import DeScramble from '../transformers/ImageDescrambler';
+import { Exception } from '../Error';
+import { WebsiteResourceKey as W } from '../../i18n/ILocale';
+
+type APIResult<T> = {
+    data: T,
+    _metadata: {
+        total_pages: number,
+    }
+}
+
+type APIManga = {
+    id: number,
+    name: string
+}
+
+type APIChapter = {
+    id: number,
+    name: string,
+    number: string,
+    status: string,
+    pages: {
+        image_url: string,
+        drm_data: string,
+        status: string
+    }[]
+}
+
+export default class extends DecoratableMangaScraper {
+    private readonly apiUrl = `${this.URI.origin}/api/v2/`;
+
+    public constructor() {
+        super('cuutruyen', 'Cứu Truyện', 'https://cuutruyen.net', Tags.Media.Manhwa, Tags.Media.Manhua, Tags.Language.Vietnamese, Tags.Source.Aggregator);
+    }
+
+    public override get Icon() {
+        return icon;
+    }
+
+    public override ValidateMangaURL(url: string): boolean {
+        return new RegExp(`^${this.URI.origin}/mangas/\\d+$`).test(url);
+    }
+
+    public override async FetchManga(provider: MangaPlugin, url: string): Promise<Manga> {
+        const mangaid = url.match(/\/mangas\/([0-9]+)/)[1];
```

Comment:
Since `ValidateMangaURL` ensures the `url` ends with the ID, this can be simplified `url.split('/').at(-1)`